avm/ptn/lz/sub-vending/modules/subResourceWrapper.bicep

```diff
@@ -2237,9 +2251,12 @@ type virtualNetworkType = {
   @description('Required. The name of the virtual network resource.')
   name: string
 
-  @description('Required. The address prefixes for the virtual network.')
+  @description('Required. An Array of 1 or more IP Address Prefixes for the Virtual Network in CIDR notation (e.g. `["10.0.0.0/16"]`) OR a single IPAM pool resource ID. When specifying an IPAM pool resource ID, you must also set ipamPoolNumberOfIpAddresses.')
   addressPrefixes: array
 
+  @description('Optional. The number of IP addresses to allocate from the IPAM pool. Required when addressPrefixes contains an IPAM pool resource ID.')
+  ipamPoolNumberOfIpAddresses: string?
```

Comment:
Just a minor one, but this only accepts specific numbers to the best of my knowledge. The number of IPs in a CIDR range. Not sure if er need to call that out?